XamlControlsGallery/Navigation/NavigationRootPage.xaml.cs

```diff
@@ -136,16 +135,49 @@ public bool CheckNewControlSelected()
             return _newControlsMenuItem.IsSelected;
         }
 
+        public void EnsureNavigationSelection(string id)

```

Comment:
Why do we need this? Are we relying on the selection to get the correct URI?

Reply:
> Are we relying on the selection to get the correct URI?

No -- we are using the navigation parameters to guarantee that the NavigationView visual state matches the right sample page content that is displayed from protocol activations.

The original protocol activation code has a few issues with handling navigation to ItemPages, particularly on cold launch. Depending on whether you are navigating to a group or an item, you can reliably get into a bad state where the NavigationView visual state does not match the page content that is displayed.